rsync_to_rdisc.py

```diff
@@ -169,95 +176,89 @@ def check_if_file_missing(required_files, input_folder, client):
     return missing
 
 
-def action_if_file_missing(folder, remove_run_file, missing, run, folder_location, run_file):
+def action_if_file_missing(transfer_settings, remove_run_file, missing, run, folder_location, run_file):
     # Send a mail and lock datatransfer
-    if not isinstance(folder.get('continue_without_email', None), bool):
-        reason = ("Unknown status {0}: {1} in settings.py for {2}").format(
-            'continue_without_email', folder.get('continue_without_email', None), folder)
+    if not isinstance(transfer_settings.get('continue_without_email', None), bool):
+        reason = "Unknown status {0}: {1} in settings.py for {2}".format(
+            'continue_without_email',
+            transfer_settings.get('continue_without_email', None),
+            transfer_settings['name']
+        )
         send_mail_incomplete(run, "settings", reason, run_file)
         return False
-    elif 'continue_without_email' in folder and folder["continue_without_email"]:
-        # Do not send a mail and do not lock datatransfer
+    # Do not send a mail and do not lock datatransfer
+    elif 'continue_without_email' in transfer_settings and transfer_settings["continue_without_email"]:
         return remove_run_file
-    elif 'continue_without_email' in folder and not folder["continue_without_email"]:
-        # Send a mail and lock datatransfer
+    # Send a mail and lock datatransfer
+    elif 'continue_without_email' in transfer_settings and not transfer_settings["continue_without_email"]:
         reason = (
             "Analysis not complete (file(s) {0} missing). "
             "Run = {1} in folder {2} ".format(" and ".join(missing), run, folder_location))
         send_mail_incomplete(run, "transfer_notcomplete", reason, run_file)
         return False
 
 
-def rsync_server_remote(
-        hpc_server, client, to_be_transferred, run_file, log=settings.log, bgarray=settings.bgarray, wkdir=settings.wkdir
-):
+def rsync_server_remote(hpc_server, client, to_be_transferred, mount_path, run_file):
```

Comment:
I added the log, bgarray and wkdir input args on purpose. 
Without the unit test fixes, I can't say if I prefer it with args or by mocking the settings object... Let me know if you fixed the unit tests so I can take a look :)

Reply:
Fixed it :)